[WebGPU] Several limits in HardwareCapabilities.mm are not realistic

mwyrzykowski opened the pull request:
#### cbc83e590816bce3fbd0bd853d64a27b01f83427
<pre>
[WebGPU] Several limits in HardwareCapabilities.mm are not realistic
<a href="https://bugs.webkit.org/show_bug.cgi?id=265853">https://bugs.webkit.org/show_bug.cgi?id=265853</a>
&lt;radar://119171264&gt;

Reviewed by Dan Glastonbury.

Reporting such large of a limit will not work in practice as
we won&apos;t be able to allocate sufficient memory if a website
attempts to approach the reported limit, so reduce the limit
to something large yet reasonable.

* Source/WebGPU/WebGPU/HardwareCapabilities.mm:
(WebGPU::largeReasonableLimit):
(WebGPU::apple3):
(WebGPU::apple4):
(WebGPU::apple5):
(WebGPU::apple6):
(WebGPU::apple7):
(WebGPU::mac2):
(WebGPU::maxUInt32Limit): Deleted.

Canonical link: <a href="https://commits.webkit.org/271776@main">https://commits.webkit.org/271776@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/45894076bd28d25420e8dc8f6fa77155238ef282

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28936 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/7597 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30292 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/31551 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26371 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/9712 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4924 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26399 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29208 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6269 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24840 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5463 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5583 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/32888 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26461 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26297 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31838 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5566 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3738 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29621 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7208 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7030 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6051 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6087 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebGPU/WebGPU/HardwareCapabilities.mm:
```diff
@@ -34,10 +34,10 @@
 {
     // FIXME: https://bugs.webkit.org/show_bug.cgi?id=252873
-    return std::numeric_limits<uint32_t>::max() - 1;
+    return 100000;
```
Does this require a comment stating it was pulled out of thin air?

mwyrzykowski replied:
I kind of think the new function name indicates that via the first part of the name `largeReasonable`

djg reviewed (approved):
:shipit:

mwyrzykowski committed:
[WebGPU] Several limits in HardwareCapabilities.mm are not realistic
https://bugs.webkit.org/show_bug.cgi?id=265853
<radar://119171264>

Reviewed by Dan Glastonbury.

Reporting such large of a limit will not work in practice as
we won't be able to allocate sufficient memory if a website
attempts to approach the reported limit, so reduce the limit
to something large yet reasonable.

* Source/WebGPU/WebGPU/HardwareCapabilities.mm:
(WebGPU::largeReasonableLimit):
(WebGPU::apple3):
(WebGPU::apple4):
(WebGPU::apple5):
(WebGPU::apple6):
(WebGPU::apple7):
(WebGPU::mac2):
(WebGPU::maxUInt32Limit): Deleted.

Canonical link: https://commits.webkit.org/271776@main
